src/client/httpClient.ts

```diff
@@ -249,18 +244,39 @@ export const handleIfErrorResponse = (response: any, data: Record<string, any>)
 
 function serializeCommand(data: Record<string, any>, pretty?: boolean): string {
   if (pretty) {
-    return JSON.stringify(data, function(key, value) {
-      if (typeof value === 'bigint') {
+    return EJSON.stringify(data, function (key, value) {
+      if (value != null && typeof value === 'bigint') {
         return Number(value);
+      } else if (value != null && typeof value === 'object') {
+        // ObjectId to strings
+        if (value.$oid) return value.$oid;
+        else if (value.$date) {
+          // Use numbers instead of strings for dates
+          value.$date = new Date(value.$date).valueOf();
+        }
       }
       return value;
     }, '  ');
   }
-  return JSON.stringify(data, function(key, value) {
-    if (typeof value === 'bigint') {
+  return EJSON.stringify(data, function (key, value) {
```

Comment:
Can you refactor out `function (key, value) { ... }` into a separate function so we avoid the duplicated logic please? I know I originally wrote this with duplicate functions, but now that the serialization logic is getting more complex this duplication is starting to look bad.

Reply:
Fixed in this commit [3b0349e](https://github.com/stargate/stargate-mongoose/pull/91/commits/3b0349e4a0b09cd2d8fb2791a9a34f573d98817f)